test/test_reductions.py

```diff
@@ -3128,21 +3128,34 @@ def test_tensor_compare_ops_empty(self, device):
             # Check if function raises error on specified zero'd dimension as reduction dim.
             self.assertRaisesRegex(IndexError, "Expected reduction dim", lambda: fn(master_input, dim=1))
 
-    # Test to ensure that the reduction operators, amax and amix throws errors for zero-dim
+    # Tests to ensure that the reduction operators, amax and amix throws errors for zero-dim
     # tensors (i.e. empty tensors)
-    def test_amax_amin_no_dim_arg(self, device):
+    def test_amax_amin_zero_dim_tensors(self, device):
         shape = (2, 0, 4)
+
         master_input = torch.randn(shape, device=device)
+        numpy_input = np.empty(shape)
         test_functions = [
-            (torch.amax, {'dtype': torch.int64}),
-            (torch.amin, {'dtype': torch.int64})
+            (torch.amax, np.amax),
+            (torch.amin, np.amin)
         ]
 
-        err_msg = "Specify the reduction dim with the 'dim' argument."
+        err_msg = "Specify the reduction dim with the 'dim' argument"
+
+        for fn, numpy_fn in test_functions:
 
-        for fn, dtype in test_functions:
+            # Cases that raises error (i.e. if dim arg is not provided)
             with self.assertRaisesRegex(RuntimeError, err_msg):
                 fn(master_input)
+ 
+            # Tests to check if reduction happens along the specified dim.
+            self.assertEqual(torch.empty((2, 0), device=device), fn(master_input, dim=2))
+            self.assertEqual(numpy_fn(numpy_input, axis=2),
```

Comment:
I think `self.assertEqual` can compare a tensor and numpy array. So you probably won't have to deal with converting tensor to numpy array or reverse.

Reply:
Yes, that works! removed it

Reply:
Also if you get `numpy_input` from copy of `tensor_input_input` you won't have to set `exact_dtype=False` (as numpy by default creates array of np.float64  while pytorch will create of torch.float).

Eg
```python
t = torch.randn(shape)
numpy_inp = t.clone().cpu().numpy()
```

Reply:
Do we really need to use `clone`? The following snippet works well:

```python
t = torch.randn(shape)
numpy_inp = t.cpu().numpy()
```

Reply:
Without `clone`, `numpy_inp` and `t` will share the same storage. So I added `clone` to be extra safe (but given that following ops don't mutate the input, without `clone` should also be ok).

Reply:
Got it! :)